parachains/runtimes/assets/statemine/src/xcm_config.rs

```diff
@@ -143,17 +149,122 @@ match_types! {
 	};
 }
 
+/// A call filter for the XCM Transact instruction. This is a temporary measure until we properly
+/// account for proof size weights.
+///
+/// Calls that are allowed through this filter must:
+/// 1. Have a fixed weight;
+/// 2. Cannot lead to another call being made;
+/// 3. Have a defined proof size weight, e.g. no unbounded vecs in call parameters.
+pub struct SafeCallFilter;
+impl Contains<RuntimeCall> for SafeCallFilter {
+	fn contains(call: &RuntimeCall) -> bool {
+		#[cfg(feature = "runtime-benchmarks")]
+		{
+			if matches!(call, RuntimeCall::System(frame_system::Call::remark_with_event { .. })) {
+				return true
+			}
+		}
+
+		match call {
+			RuntimeCall::System(
+				frame_system::Call::set_heap_pages { .. } |
+				frame_system::Call::set_code { .. } |
+				frame_system::Call::set_code_without_checks { .. } |
+				frame_system::Call::kill_prefix { .. },
+			) |
+			RuntimeCall::ParachainSystem(..) |
```

Comment:
Just a note that we'll want to test an upgrade on Westmint to ensure this passes before putting this runtime on to Statemint/e. I.e. if this is runtime `n` then we want `n+1` on Westmint before `n` on Statemint/e.